content/post/imbalanced_data.Rmd

```diff
@@ -0,0 +1,238 @@
+packages: ['mlr3', 'mlr3learners', 'mlr3pipelines', 'mlr3tuning']
+---
+
+# Intro 
+
+This use case compares different approaches to handle class imbalancy for the [`optdigits`](https://www.openml.org/d/980) binary classification data set using the mlr3 package. 
+We mainly focus on undersampling the majority class, oversampling the minority class, and the smote imbalancy correction that enriches the minority class with synthetic data. 
+The use case requires prior knowledge in basic ML concepts (e.g., issues imbalanced data, hyperparameter tuning, nested cross-validation) and `mlr3`, `mlr3pipelines` and `mlr3tuning` is required.
+You can find most of the content here also in the [mlr3book](https://mlr3book.mlr-org.com/) in a more detailed way.
+
+The steps are performed:
+
+* Retrieve data sets from `OpenML`
+* Define imbalancy correction pipelines (undersampling, oversampling and smote) with `mlr3pipelines`  
+* Combine pipeline with a learner and autotune it with `mlr3tuning`
+* Benchmark the autotuned pipelines and visualize the results using `mlr3viz`
+
+# Prerequisites
+
+## Loading required packages
+
+```{r}
+library("mlr3")          # mlr3 base package
+library("mlr3misc")      # contains some helper functions
+library("mlr3measures")  # lots of performance measures
+library("mlr3pipelines") # create ML pipelines
+library("mlr3tuning")    # tuning ML algorithms
+library("mlr3learners")  # additional ML algorithms
+library("mlr3viz")       # autoplot for benchmarks
+library("paradox")       # hyperparameter space
+library("OpenML")        # to obtain data sets
+library("smotefamily")   # smote algorithm for imbalancy correction
+```
+
+## Retrieve data sets from OpenML
+
+OpenML.org is an open machine learning platform, which allows users to share data, code and machine learning experiments.
+The OpenML R package can query available data sets and download them directly from within an R session:
+
+```{r}
+# get list of curated binary classification data sets (see https://arxiv.org/abs/1708.03731v2) 
+ds = listOMLDataSets(
+  number.of.classes = 2, 
+  number.of.features = c(1, 100), 
+  number.of.instances = c(5000, 10000)
+)
+# select imbalanced data sets (without categorical features as smote cannot handle them)
+ds = subset(ds, minority.class.size/number.of.instances < 0.2 & number.of.symbolic.features == 1)
+ds
+
+# pick one data set from list above 
+d = getOMLDataSet(980)
+d
+```
+
+After downloading the chosen data set, we create an `mlr3` classification task:
+
+```{r}
+# make sure target is a factor and create mlr3 tasks 
+data = as.data.frame(d)
+data[[d$target.features]] = as.factor(data[[d$target.features]])
+task = TaskClassif$new(id = d$desc$name, backend = data, target = d$target.features)
+task
+```
+
+Quick overview of the data:
+```{r}
+skimr::skim(data)
+```
+
+## Imbalancy correction
+
+In `mlr3pipelines`, there is a `classbalancing` and a `smote` pipe operator that can be combined with any learner. 
```

Comment:
Consider adding links when mentioning for the post relevant functions 

In `mlr3pipelines`, there is a [`classbalancing`](https://mlr3pipelines.mlr-org.com/reference/mlr_pipeops_classbalancing.html) and a [`smote` ](https://mlr3pipelines.mlr-org.com/reference/mlr_pipeops_smote.html) pipe operator that can be combined with any learner.